### PR TITLE
CI: Remove memo program from downstream job

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

The SPL programs are in the process of being moved to program-specific repos. Memo now exists in https://github.com/solana-program/memo, and there's a PR to remove it from SPL at
https://github.com/solana-labs/solana-program-library/pull/7535. If we merge that, however, the downstream jobs will immediately fail.

#### Summary of changes

Remove all references to the memo program in downstream CI.

### CODE REVIEW

**Comment by joncinque** [comment]
Blocked until #3806 lands

**Review by yihau — APPROVED** [review]

**Pull request merged**
